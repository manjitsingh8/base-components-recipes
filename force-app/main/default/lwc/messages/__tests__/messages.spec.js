/*
 * Copyright (c) 2019, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

import { createElement } from 'lwc';
import Element from 'c/messages';

const createMessages = () => {
    const element = createElement('c-messages', { is: Element });
    document.body.appendChild(element);
    return element;
};

describe('c-messages', () => {
    it('default', () => {
        const elem = createMessages();
        elem.setError(getErrorResponse([{ errorCode: 'foo', message: 'bar' }]));

        return Promise.resolve().then(() => {
            expect(elem).toMatchSnapshot();
        });
    });
    it('just a message', () => {
        const elem = createMessages();
        elem.setError(getErrorResponse([{ message: 'bar' }]));

        return Promise.resolve().then(() => {
            expect(elem).toMatchSnapshot();
        });
    });
    it('has no error', () => {
        const elem = createMessages();
        elem.setError(null);

        return Promise.resolve().then(() => {
            expect(elem).toMatchSnapshot();
        });
    });
});

function getErrorResponse(body) {
    return {
        ok: false,
        status: 400,
        statusText: 'Bad Request',
        body
    };
}
