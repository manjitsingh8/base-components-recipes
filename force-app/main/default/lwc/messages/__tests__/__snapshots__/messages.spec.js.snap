// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-messages default 1`] = `
<c-messages>
  #shadow-root(open)
    <div
      class="slds-notify_container slds-is-relative"
    >
      <div
        class="slds-notify slds-notify_toast slds-theme_error"
        role="alert"
      >
        <div
          class="slds-notify__content"
        >
          <h2
            class="slds-text-heading_small"
          >
            bar
          </h2>
          <p>
            foo
          </p>
        </div>
      </div>
    </div>
</c-messages>
`;

exports[`c-messages has no error 1`] = `
<c-messages>
  #shadow-root(open)
</c-messages>
`;

exports[`c-messages just a message 1`] = `
<c-messages>
  #shadow-root(open)
    <div
      class="slds-notify_container slds-is-relative"
    >
      <div
        class="slds-notify slds-notify_toast slds-theme_error"
        role="alert"
      >
        <div
          class="slds-notify__content"
        >
          <h2
            class="slds-text-heading_small"
          >
            bar
          </h2>
        </div>
      </div>
    </div>
</c-messages>
`;
